src/inmanta/server/agentmanager.py

```diff
@@ -213,109 +206,167 @@ async def _unpause_agent(self, env: data.Environment, endpoint: Optional[str] =
                 live_session = self.tid_endpoint_to_session.get(key)
                 # If the agent has a live_session, the agent wasn't paused
                 if not live_session:
-                    session = self._get_session_for(tid=env.id, endpoint=agent_name)
+                    session = self.get_session_for(tid=env.id, endpoint=agent_name)
                     if session:
                         self.tid_endpoint_to_session[key] = session
                         await session.get_client().set_state(agent_name, enabled=True)
 
-    # Agent Management
-    async def ensure_agent_registered(self, env: data.Environment, nodename: str) -> data.Agent:
+    async def _process_session_listener_actions(self) -> None:
+        """
+            This is the consumer of the _session_listener_actions queue.
+        """
+        while not self.is_stopping():
+            try:
+                session_action = await self._session_listener_actions.get()
+                await self._process_action(session_action)
+            except asyncio.CancelledError as e:
+                # Cancel this task
+                raise e
```

Comment:
I added the raise because of the following sentence in the documentation:

`A task will be marked as cancelled when the wrapped coroutine terminates with a CancelledError exception (even if cancel() was not called).`

So if we would catch the CancelledError exception, cancelled() would never be True, while the cancel() method was invoked. This sounds weird to me. On the other hand. If we keep it this way, we will get a log line "Task was cancelled" every time the server shuts down. Indeed, it's better to add a return.